### PR TITLE
Qmstr should work with no packages

### PR DESCRIPTION
**Pull request by GiasemiSh**
Since we changed the structure to include a project node, we had not changed the implementation of using multiple packages. 
Now, GetPackageNode returns a stream of packages and QMSTR doesn't fail if no packages exist.

closes https://github.com/QMSTR/qmstr/issues/274 https://github.com/QMSTR/qmstr/issues/275

### CODE REVIEW

File: modules/analyzers/git-analyzer/main.go [hunk 1]
```diff
@@ -26,6 +28,11 @@ type Revision struct {
 }
 
 type GitAnalyzer struct {
+	workdirs []string
+	pkgInfo  packageInfo
+}
+
+type packageInfo struct {
```
Comment (hemarkus) [inline modules/analyzers/git-analyzer/main.go]: Why do we need a new struct here? Also the struct's name is totally misleading. This has nothing to do with a package when it holds only revision stuff.

File: lib/go-qmstr/cli/common.go [hunk 2]
```diff
@@ -47,6 +47,18 @@ func getUniqueFileNode(fnode *service.FileNode) (*service.FileNode, error) {
 	return fileNode, nil
 }
 
+func getUniquePackageNode(pnode *service.PackageNode) (*service.PackageNode, error) {
+	stream, err := controlServiceClient.GetPackageNode(context.Background(), pnode)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get package node: %v", err)
+	}
+	pkgNode, err := stream.Recv()
```
[the remark this answers is not in the document]
Comment (GiasemiSh) [reply lib/go-qmstr/cli/common.go]: No need to close this stream. 
This stream is on the client side and just receives the packages that are send to it.

[the remark this answers is not in the document]
Comment (hemarkus) [reply lib/go-qmstr/cli/common.go]: Exactly, you are on client-side consuming only one message from the stream. But what about the server side? It waits for you to consume the rest of the stream. See: https://github.com/grpc/grpc-go/commit/365770fcbd7dfb9d921cb44827ede770f33be44f

[the remark this answers is not in the document]
Comment (GiasemiSh) [reply lib/go-qmstr/cli/common.go]: I see..

**Pull request merged**